### ISSUE TITLE
Feature request: Rewrite rule for "function argument with parens and braces"

### ISSUE THREAD

**Issue by Daenyth** [scalameta/scalafmt#1321]
I'd like scalafix to be able to detect when a single-argument parameter group where the parameter is a function (things like `map`, `flatMap`, etc) have redundant parens around function braces.

```
//before
foo.map(x => { blah(x) })
//after
foo.map { x => blah(x) }
```

and
```
//before
foo.map({x => blah(x) })
//after
foo.map { x => blah(x) }
```

**Comment by olafurpg** [comment]
Thanks for reporting! I welcome contributions adding this functionality, please reopen if you would like to take a stab at implementing this.

**Comment by Daenyth** [comment]
@olafurpg Can you give some guidance on where I'd start if I wanted to implement this myself?

**Comment by kitbellew** [comment]
@Daenyth the first of your request is a duplicate of #1237 which has already been fixed in #1563 and released in https://github.com/scalameta/scalafmt/releases/tag/v2.3.0-RC2.

I have fixed the second part and will submit the patch today.
